install.sh

```diff
@@ -0,0 +1,504 @@
+#!/usr/bin/env sh
+# Modified version of: https://github.com/starship/starship/blob/master/install/install.sh
+
+set -eu
+printf '\n'
+
+BOLD="$(tput bold 2>/dev/null || printf '')"
+GREY="$(tput setaf 0 2>/dev/null || printf '')"
+UNDERLINE="$(tput smul 2>/dev/null || printf '')"
+RED="$(tput setaf 1 2>/dev/null || printf '')"
+GREEN="$(tput setaf 2 2>/dev/null || printf '')"
+YELLOW="$(tput setaf 3 2>/dev/null || printf '')"
+BLUE="$(tput setaf 4 2>/dev/null || printf '')"
+MAGENTA="$(tput setaf 5 2>/dev/null || printf '')"
+NO_COLOR="$(tput sgr0 2>/dev/null || printf '')"
+
+SUPPORTED_TARGETS="linux_386 linux_amd64 linux_arm64 linux_armv6 \
+                   darwin_amd64 darwin_arm64 \
+                   windows_amd64"
+
+info() {
+  printf '%s\n' "${BOLD}${GREY}>${NO_COLOR} $*"
+}
+
+warn() {
+  printf '%s\n' "${YELLOW}! $*${NO_COLOR}"
+}
+
+error() {
+  printf '%s\n' "${RED}x $*${NO_COLOR}" >&2
+}
+
+completed() {
+  printf '%s\n' "${GREEN}✓${NO_COLOR} $*"
+}
+
+has() {
+  command -v "$1" 1>/dev/null 2>&1
+}
+
+curl_is_snap() {
+  curl_path="$(command -v curl)"
+  case "$curl_path" in
+  /snap/*) return 0 ;;
+  *) return 1 ;;
+  esac
+}
+
+# Make sure user is not using zsh or non-POSIX-mode bash, which can cause issues
+verify_shell_is_posix_or_exit() {
+  if [ -n "${ZSH_VERSION+x}" ]; then
+    error "Running installation script with \`zsh\` is known to cause errors."
+    error "Please use \`sh\` instead."
+    exit 1
+  elif [ -n "${BASH_VERSION+x}" ] && [ -z "${POSIXLY_CORRECT+x}" ]; then
+    error "Running installation script with non-POSIX \`bash\` may cause errors."
+    error "Please use \`sh\` instead."
+    exit 1
+  else
+    true # No-op: no issues detected
+  fi
+}
+
+get_tmpfile() {
+  suffix="$1"
+  if has mktemp; then
+    printf "%s.%s" "$(mktemp)" "${suffix}"
+  else
+    # No really good options here--let's pick a default + hope
+    printf "/tmp/tasker.%s" "${suffix}"
+  fi
+}
+
+# Test if a location is writable by trying to write to it. Windows does not let
+# you test writability other than by writing: https://stackoverflow.com/q/1999988
+test_writeable() {
+  path="${1:-}/test.txt"
```

Comment:
should we use random filename? so it won't remove the preexisting test.txt file later